metricbeat/module/system/core/_meta/fields.yml

```diff
@@ -107,3 +107,29 @@
         The amount of CPU time spent in involuntary wait by the virtual CPU while the hypervisor
         was servicing another processor.
         Available only on Unix.
+
+    - name: model_number
+      type: keyword
+      description: >
+        CPU model number. Only availabe on Linux
+
+    - name: model_name
+      type: keyword
+      description: >
+        CPU model name. Only availabe on Linux
+
+    - name: mhz
+      type: float
+      description: >
+        CPU core current clock. Only availabe on Linux
+
+    - name: core_id
+      type: integer
```

Comment:
Are you sure it is always integer? Does it make sense to make it an integer, e.g. are users going to search for IDs that are bigger than X?

I have seen weird behaviour in systemd where an ID that used to be an integer, changed to a string. Making numerical IDs keyword seems more futureproof.

Reply:
No, I'm not sure. I looked a lot but did not find any proper documentation about the format of `/proc/cpuinfo` :/

I can only think of stupid ways to use it as an integer, I'll change to to keyword, its indeed more futureproof.

Reply:
I also changed the `physical_id`.